### DISCUSSION TITLE
get data of OutputArgument Node

### DISCUSSION THREAD

**Discussion by UlPix** [FreeOpcUa/opcua-asyncio#1439] [Q&A] [answered]
Im fairly new to OPCUA, but I'm trying to read the value of an OutputArgument Node from a called method that just multiplies two numbers.
I found  [this](https://github.com/FreeOpcUa/opcua-asyncio/discussions/380) ,however it didnt answer exactly how to read the value of that node.
In UAExpert my attributes of the OutputArguments look exactly the same as  [this](https://github.com/FreeOpcUa/opcua-asyncio/discussions/380#discussioncomment-254102), but in my case, `client.get_node("ns=2;i=4").read_value()` is returning 
[Argument(Name=None, DataType=NodeId(Identifier=11, NamespaceIndex=0, NodeIdType=<NodeIdType.TwoByte: 0>), ValueRank=-1, ArrayDimensions=[], Description=LocalizedText(Locale=None, Text=None))] 

How can I directly access/read the value of that OutputArgument node?

**Comment by schroeder-** [comment]
This is the value. If you want to know how the datatype is named you can read it like this:
```Python
node = client.get_node("ns=2;i=4")
# or if you just know the node of the method
meth_node = client.get_node("...")
node = await meth_node.get_child("OutputArgument")

args = await node.read_value()
print("Args")
for arg in args:
     arg_node = client.get_node(arg.DataType)
     display_name = await arg_node.read_display_name()
     browse_name = await arg_node.read_browse_name()
     print(f"{display_name}:{arg.ValueRank}")
```

**Reply by UlPix** [reply]
But there is no way I can access the actual value of the (in my case multiplication) output node other than calling the method on the server?

What I'd eventually like to achieve is subscribing to the datachange of the (multiplicated) value of my OutputArgument Node, so I thought once I had the OutputArgument as a result of the multiplication method, I could subcribe to that change.

**Reply by schroeder-** [reply, marked as answer]
This is not possible in opc ua.
